addons/point_of_sale/models/pos_session.py

```diff
@@ -178,6 +178,13 @@ def _check_start_date(self):
             if (company.period_lock_date and start_date <= company.period_lock_date) or (company.fiscalyear_lock_date and start_date <= company.fiscalyear_lock_date):
                 raise ValidationError(_("You cannot create a session before the accounting lock date."))
 
+    def _check_bank_statement_state(self):
+        for record in self:
```

Comment:
you can either rename "for session in self", or just use self and no loop as the function 'action_pos_session_close' seems to work as a self.ensure_one() function